File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cpp
class Solution {
public:

    void reverseString(string & s) {
        int n = s.size();
        int start = 0;
        int end = n-1;

        while(start<end){
            swap(s[start],s[end]);
            start++;
            end--;
        }
                
    }
    string reverseWords(string s) {
        int n = s.length();

        string ans = "";


        for(int i = 0; i<n ; i++){
            string word = "";
            while(i < n && s[i] != ' '){
                word += s[i];
                i++;
            }
            if(word.length() > 0){
                reverseString(word);
                ans += ' ' + word;
            }

        }
        return ans.substr(1);

        
    }
};

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.java
class Solution {
    public void revWords(int a , int b, char[] S){
        while(a < b){
            char temp = S[a];
            S[a] = S[b];
            S[b] = temp;
            a++;
            b--;
        }
    }
    public String reverseWords(String s) {
        char[] S = s.toCharArray();
        int n = s.length();
        int i = 0;
        int j = 0;
        
        while(j < n){
            while(j < n && S[j] != ' '){
                j++;
            }
            revWords(i,j-1,S);
            j++;
            i=j;
        }
        return new String(S);
    }
}